grafana/grafana
# Tempo: Add custom headers middleware for grpc client

**What is this feature?**

This adds middleware that passes the custom headers defined in the data source config to the grpc client. This way users can still pass headers like `X-Scope-Orgid` when using streaming with Tempo.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:



Fixes #

**Special notes for your reviewer:**
To test:
- Build docker image from main https://github.com/grafana/xk6-client-tracing with `make docker`
- Run this docker compose https://github.com/grafana/tempo/tree/main/example/docker-compose/multi-tenant with grafana commented out and add or modify data sources based on the grafana-datasources.yaml
- Test that you can see data (with previous version or without the `X-Scope-Orgid` header you should get an error)

Please check that:
